PRODUCTS DATASET


--- Iniciando as atividades de tratamento do DF importamos as bibliotecas pandas e numpy, através delas poderemos utilizar comandos que facilitarão a limpeza dos dados e organização do DF para posteriormente iniciar as análises. 
   Neste mesmo dataframe de importação das bilbiotecas criamos uma variável chamada DTS, ela contém na memória o arquivo CSV e quando necessitamos referenciar o arquivo utilizamos a nomenclatura da variável(DTS) juntamente com os scripts das bilbiotecas para executar as ações no DF e assim realizar as operações necessárias.  

In [2]:
import pandas as pd 
import numpy as np


dts = pd.read_csv("C:/Users/leopo/OneDrive/Documentos/KORU IFOOD POTENCIA TECH/MOD3 BIG DATA PYTHON E PYSPARK/download_datasets/brazilian-ecommerce/olist_products_dataset.csv")

- Referenciando DTS podemos visualizar parte do arquivo contendo as colunas e dados. Por se tratar de um dataset grande não é possível e nem recomendado visualizar todo o arquivo mas apenas seu head e as primeiras linhas para ter noção de como o arquivo está composto.

In [14]:
dts

product_id              product_category_name  \
0      1e9e8ef04dbcff4541ed26657ea517e5                         perfumaria   
1      3aa071139cb16b67ca9e5dea641aaa2f                              artes   
2      96bd76ec8810374ed1b65e291975717f                      esporte_lazer   
3      cef67bcfe19066a932b7673e239eb23d                              bebes   
4      9dc1a7de274444849c219cff195d0b71              utilidades_domesticas   
...                                 ...                                ...   
32946  a0b7d5a992ccda646f2d34e418fff5a0                   moveis_decoracao   
32947  bf4538d88321d0fd4412a93c974510e6  construcao_ferramentas_iluminacao   
32948  9a7c6041fa9592d9d9ef6cfe62a71f8c                    cama_mesa_banho   
32949  83808703fc0706a22e264b9d75f04a2e             informatica_acessorios   
32950  106392145fca363410d287a815be6de4                    cama_mesa_banho   

       product_name_lenght  product_description_lenght  product_photos_qty  \
0                     40.0                       287.0                 1.0   
1                     44.0                       276.0                 1.0   
2                     46.0                       250.0                 1.0   
3                     27.0                       261.0                 1.0   
4                     37.0                       402.0                 4.0   
...                    ...                         ...                 ...   
32946                 45.0                        67.0                 2.0   
32947                 41.0                       971.0                 1.0   
32948                 50.0                       799.0                 1.0   
32949                 60.0                       156.0                 2.0   
32950                 58.0                       309.0                 1.0   

       product_weight_g  product_length_cm  product_height_cm  \
0                 225.0               16.0               10.0   
1                1000.0               30.0               18.0   
2                 154.0               18.0                9.0   
3                 371.0               26.0                4.0   
4                 625.0               20.0               17.0   
...                 ...                ...                ...   
32946           12300.0               40.0               40.0   
32947            1700.0               16.0               19.0   
32948            1400.0               27.0                7.0   
32949             700.0               31.0               13.0   
32950            2083.0               12.0                2.0   

       product_width_cm  
0                  14.0  
1                  20.0  
2                  15.0  
3                  26.0  
4                  13.0  
...                 ...  
32946              40.0  
32947              16.0  
32948              27.0  
32949              20.0  
32950               7.0  

[32951 rows x 9 columns]

- O script abaixo exibe através do comando .dtypes a composição das colunas onde "objects" são strings e "float64" trata-se de números de pontos flutuantes ou com casas decimais após a vírgula. As colunas compostas por "float64" podem ser tratadas com operadores matemáticos no momento da análise exploratória, por tanto, caso alguma coluna contendo números que necessitarão sofrer operações nas análises não esteja no formato "float64" será necessário a conversão para este formato para a efetividade das análises.

In [15]:
dts.dtypes

product_id                     object
product_category_name          object
product_name_lenght           float64
product_description_lenght    float64
product_photos_qty            float64
product_weight_g              float64
product_length_cm             float64
product_height_cm             float64
product_width_cm              float64
dtype: object

- O script abaixo exibe o total de linhas e colunas presentes no DF utilizando o comando ".shape" ,  neste caso podemos observar que ele é composto por 32951 linhas e 9 colunas

In [16]:
dts.shape

(32951, 9)

- O script abaixo exibe o nome das colunas presentes no arquivo. Através do nome das colunas poderemos elaborar scripts especificando quais colunas serão tratadas.

In [17]:
dts.columns

Index(['product_id', 'product_category_name', 'product_name_lenght',
       'product_description_lenght', 'product_photos_qty', 'product_weight_g',
       'product_length_cm', 'product_height_cm', 'product_width_cm'],
      dtype='object')

A coluna PRODUCT_CATEGORY_NAME é composta por valores em lowercase sem caracteres especiais e os espaços estão preenchidos por underscore. 

- A coluna abaixo foi tratada excluindo caracteres especiais, espaços ou letras mauiúsculas. Uma das formas de tratar um df para análise é padronizar nomes  em lowercase ou uppercase, preencher espaços com underscore e remover como dito caracteres especiais como acentos, vírgulas, hastags e etc. 
  Sem o devido cuidado na padronização nos nomes poderemos obter resultados contraditórios, um exemplo é a mesma palavra uma composta por "ç" e outra sem, no momento da busca dos dados o sistema pode compreender que a palavra difere da ação contante no script e não trazer os valores contidos na mesma gerando resultados contraditórios, errôneos ou incompletos. 

In [20]:
dts[['product_category_name']]

product_category_name
0                             perfumaria
1                                  artes
2                          esporte_lazer
3                                  bebes
4                  utilidades_domesticas
...                                  ...
32946                   moveis_decoracao
32947  construcao_ferramentas_iluminacao
32948                    cama_mesa_banho
32949             informatica_acessorios
32950                    cama_mesa_banho

[32951 rows x 1 columns]

- O script abaixo realiza a contagem dos valores contidos na coluna "product_category_name", temos então 32341 Linhas e o tipo dos dados é  "int64" os dados deste tipo são números inteiros reais sem acréscimo de valores após a vírgula, sendo assim, poderemos executar análises com base em operadores matemáticos e estatísticos ára as análises. 

In [22]:
dts[['product_category_name']].count()

product_category_name    32341
dtype: int64

- O script abaixo nos retorna através do comando ".isna" se há a presença de valores nulos na coluna. Esta função nos retorna um valor booleano "TRUE" or "FALSE". Sendo TRUE a confirmação da existência de valores nulos que dependendo do tipo de análise e contexto poderemos optar pela exclusão ou  preenchimento destes nulos com valores do tipo "PREENCHIDO" nos sinlaizando de forma clara e objetiva que naquele campo havia um valor Nulo e que foi tratado. 

In [24]:
dts.isna()

product_id  product_category_name  product_name_lenght  \
0           False                  False                False   
1           False                  False                False   
2           False                  False                False   
3           False                  False                False   
4           False                  False                False   
...           ...                    ...                  ...   
32946       False                  False                False   
32947       False                  False                False   
32948       False                  False                False   
32949       False                  False                False   
32950       False                  False                False   

       product_description_lenght  product_photos_qty  product_weight_g  \
0                           False               False             False   
1                           False               False             False   
2                           False               False             False   
3                           False               False             False   
4                           False               False             False   
...                           ...                 ...               ...   
32946                       False               False             False   
32947                       False               False             False   
32948                       False               False             False   
32949                       False               False             False   
32950                       False               False             False   

       product_length_cm  product_height_cm  product_width_cm  
0                  False              False             False  
1                  False              False             False  
2                  False              False             False  
3                  False              False             False  
4                  False              False             False  
...                  ...                ...               ...  
32946              False              False             False  
32947              False              False             False  
32948              False              False             False  
32949              False              False             False  
32950              False              False             False  

[32951 rows x 9 columns]

In [3]:
dts.isnull()

product_id  product_category_name  product_name_lenght  \
0           False                  False                False   
1           False                  False                False   
2           False                  False                False   
3           False                  False                False   
4           False                  False                False   
...           ...                    ...                  ...   
32946       False                  False                False   
32947       False                  False                False   
32948       False                  False                False   
32949       False                  False                False   
32950       False                  False                False   

       product_description_lenght  product_photos_qty  product_weight_g  \
0                           False               False             False   
1                           False               False             False   
2                           False               False             False   
3                           False               False             False   
4                           False               False             False   
...                           ...                 ...               ...   
32946                       False               False             False   
32947                       False               False             False   
32948                       False               False             False   
32949                       False               False             False   
32950                       False               False             False   

       product_length_cm  product_height_cm  product_width_cm  
0                  False              False             False  
1                  False              False             False  
2                  False              False             False  
3                  False              False             False  
4                  False              False             False  
...                  ...                ...               ...  
32946              False              False             False  
32947              False              False             False  
32948              False              False             False  
32949              False              False             False  
32950              False              False             False  

[32951 rows x 9 columns]

- FINALIZAÇÃO  após as análises do df podemos concluir o arquivo csv e caso necessário aplicar as alterações necessárias para gerar as análises futuras com dados de qualidade e organizados de forma que facilite a extração e uso dos mesmos de acordo com as regras de negócio, o contexto em que serão empregados estas informações impacta diretamente na forma como o arquivo csv será tratado, portanto, entender do negócio e a finalidade de uso fará com que tome as decisões necessárias na tratativa do df.   